common/handlers/users/delete.go

```diff
@@ -0,0 +1,114 @@
+package users
+
+import (
+	"context"
+
+	dsr "github.com/aserto-dev/go-directory/aserto/directory/reader/v3"
+	dsw "github.com/aserto-dev/go-directory/aserto/directory/writer/v3"
+	serrors "github.com/elimity-com/scim/errors"
+	"github.com/rs/zerolog"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func (u UsersResourceHandler) Delete(ctx context.Context, id string) error {
```

Comment:
This one is pretty long too. I think the body of the `for` loop can become a separate function.